contracts/roles/RolesBase.sol

```diff
@@ -2,24 +2,68 @@ pragma solidity ^0.8.0;
 
 import {Roles} from "./Roles.sol";
 
+/**
+* @title Roles API
+* @dev This base smart contract is resposible for managing roles 
+* and offer internal functions for adding/removing roles from
+* an address.
+* Roles are denotted by a hashed string that hold a mapping of
+* address => bool
+* To avoid storage slot conflicts, the hashed string provided is
+* concatted with "roles.storage." internally.
```

Comment:
```suggestion
* @dev This base smart contract is responsible for managing roles 
* and offers internal functions for adding/removing roles from
* an address.
* Roles are denoted by a hashed string that hold a mapping of
* address => bool
* To avoid storage slot conflicts, the hashed string provided is
* concatenated with "roles.storage." internally.
```